### PR TITLE
Add FirebaseAuth identity token

### PR DESCRIPTION
**Pull request by petrpavlik**
- Add support for Firebase Auth by introducing `FirebaseAuthIdentityToken`. 
- Fix some unit tests not being run. Probably a regression cause by migration to the new testing framework.

Adding support for Firebase Authentication was okayed by @ptoffy, see here https://github.com/vapor/jwt/issues/159.

I'll also follow up with a PR for https://github.com/vapor/jwt once this is merged in.

### CODE REVIEW

**Review by ptoffy — CHANGES_REQUESTED** [review]
Thanks for this! Just missing some formatting. Feel free to run `swift format --recursive --in-place .` with https://github.com/vapor/contributing/blob/main/.swift-format in the project's directory. While you're at it could you add some names to the `@Test`s which were missing? Something similar to the other test suites should be fine

**Comment by petrpavlik** [comment]
@ptoffy done, please take another look.

Not sure though why the windows CI pass failed on some CCryptoBoringSSL (linker?) errors?

**Comment by ptoffy** [comment]
Yeah looks like this should fix it https://github.com/apple/swift-crypto/commit/4518e5e680ea71b60d5c4644939bdeebbb0301e1 but there hasn't been a release yet

File: Sources/JWTKit/Vendor/FirebaseAuthIdentityToken.swift [hunk 1]
```diff
@@ -0,0 +1,140 @@
+#if !canImport(Darwin)
+    import FoundationEssentials
+#else
+    import Foundation
+#endif
+
+public struct FirebaseAuthIdentityToken: JWTPayload {
+
```
Comment (ptoffy) [inline Sources/JWTKit/Vendor/FirebaseAuthIdentityToken.swift:8]: Nit: could you remove this whitespace along with the others to conform to the rest of the repo?

File: Sources/JWTKit/Vendor/FirebaseAuthIdentityToken.swift [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    /// Additional Firebase-specific claims
+    public struct Firebase: Codable, Sendable {
+
```
Comment (ptoffy) [inline Sources/JWTKit/Vendor/FirebaseAuthIdentityToken.swift:11]: Nit: whitespace

File: Sources/JWTKit/Vendor/FirebaseAuthIdentityToken.swift [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        self.email = email
+        self.picture = picture
+        self.name = name
+        self.emailVerified = emailVerified
+        self.phoneNumber = phoneNumber
+        self.firebase = firebase
+    }
+
+    public func verify(using _: some JWTAlgorithm) throws {
+
```
Comment (ptoffy) [inline Sources/JWTKit/Vendor/FirebaseAuthIdentityToken.swift:125]: Nit: whitespace

File: Tests/JWTKitTests/VendorTokenTests.swift [hunk 4]
```diff
@@ -255,4 +262,87 @@ struct VendorTokenTests {
             try await collection.verify(jwt, as: MicrosoftIdentityToken.self)
         }
     }
+
+    @Test("Test Firebase ID Token")
+    func testFirebaseIDToken() async throws {
+
```
Comment (ptoffy) [inline Tests/JWTKitTests/VendorTokenTests.swift:268]: Nit: whitespace

File: Tests/JWTKitTests/VendorTokenTests.swift [hunk 5]
```diff
@@ -255,4 +262,87 @@ struct VendorTokenTests {
...
+        let jwt = try await collection.sign(token)
+
+        await #expect(throws: Never.self) {
+            try await collection.verify(jwt, as: FirebaseAuthIdentityToken.self)
+        }
+    }
+
+    @Test("Test Firebase ID Token that is not from Google (invalid issuer)")
+    func testFirebaseIDTokenNotFromGoogle() async throws {
+
```
Comment (ptoffy) [inline Tests/JWTKitTests/VendorTokenTests.swift:293]: Nit: whitespace

**Review by ptoffy — CHANGES_REQUESTED** [review]
Just a few whitespace nits to be in par with the rest of the repo. I would have thought swift-format would catch these but maybe we need to be stricter 🤷

**Comment by petrpavlik** [comment]
Thanks, I'll fix those. Yeah, I just ran swift-format before pushing the changes, relying on that I'd fix everything for me.

**Review by ptoffy — APPROVED** [review]
Thanks!

**Pull request merged**